src/main/java/com/gmail/llmdlio/townyflight/TownyFlightAPI.java

```diff
@@ -199,19 +214,50 @@ public void testForFlight(Player player, boolean silent) {
 	 * and are not given a flight bypass of some kind, remove their flight. Called when
 	 * a town has their free flight disabled.
 	 */
-	public void takeFlightFromPlayersInTown(Town town) {
+	public static void takeFlightFromPlayersInTown(Town town, String cause) {
 		for (final Player player : new ArrayList<>(Bukkit.getOnlinePlayers())) {
-			if (player.hasPermission("townyflight.bypass")
+
+
+			if ((player.hasPermission("townyflight.bypass") || !player.getAllowFlight()
 				|| !player.getAllowFlight()
 				|| TownyAPI.getInstance().isWilderness(player.getLocation())
 				|| !TownyAPI.getInstance().getTown(player.getLocation()).equals(town)
-				|| TownyFlightAPI.getInstance().canFly(player, true))
+				|| TownyFlightAPI.getInstance().canFly(player, true)))
 				continue;
 
-			TownyFlightAPI.getInstance().removeFlight(player, false, true, "");
+			TownyFlightAPI.getInstance().removeFlight(player, false, true, cause);
 		}	
 	}
 
+	/**
+	 * Parse over the players online in the server and if they're in the given {@link Town},
+	 * Check if they can fly, and then add flight to them.
+	 */
+	public static void addFlightToPlayersInTown(Town town){
+		for (final Player player : new ArrayList<>(Bukkit.getOnlinePlayers())) {

```

Comment:
Instead of Bukkit.getOnlinePlayer() you can use the TownyAPI method for getting online player in a single town.